source/agora/common/crypto/Key.d

```diff
@@ -248,9 +260,13 @@ public struct PublicKey
         // The underlying function does not expose a safe interface,
         // but we know (thanks to tests and careful inspection)
         // that our data type match and they are unlikely to change
-        return 0 ==
-            crypto_sign_ed25519_verify_detached(
-                signature[].ptr, msg.ptr, msg.length, this.data[].ptr);
+        // return 0 ==
+        //     crypto_sign_ed25519_verify_detached(
+        //         signature[].ptr, msg.ptr, msg.length, this.data[].ptr);
+
+        import agora.crypto.Schnorr;
+        auto X = Point(this.data[]);
+        return agora.crypto.Schnorr.verify(X, signature, msg);
```

Comment:
Don't comment out code, remove it. Also if we're switching to `verify` then we don't need the `DataType` aliases anymore, do we ?

Reply:
Updated.
But I don't understand why this change is related to the `DataType` alias. Anyway did you mean this kind of change?

```
public struct PublicKey
{
    BitBlob!(crypto_core_ed25519_BYTES * 8) data;
    alias data this;

    /// Construct an instance from binary data
    public this (const BitBlob!(crypto_core_ed25519_BYTES * 8) args)
        pure nothrow @safe @nogc
    {
        this.data = args;
    }
```